substreams/rpccalls.go

```diff
@@ -353,8 +345,8 @@ func (e *RPCEngine) rpcDoWithRetry(
 				break
 			}
 		}
-
-		if e.LastestFallbackDuration != nil {
+		fallbackDuration := reqctx.EthCallFallbackToLatestDuration(ctx)
+		if fallbackDuration > 0 {
```

Comment:
Also, now that the fallbackDuration is not a pointer, there's no way to "activate the 'latest' mechanism on EVERY BLOCK".
You could set fallbackDuration to `1 nanosecond`, but some chains tend to produce blocks with timestamps in the future (ex: base).
I suggest `if fallbackDuration != 0`

This way we can set fallbackDuration to `- 99999h` to get behavior of setting 'latest' on very call.